code/go/internal/validator/folder_item_content.go

```diff
@@ -9,59 +9,115 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/fs"
-	"mime"
 
 	"github.com/pkg/errors"
 	"gopkg.in/yaml.v3"
+
+	ve "github.com/elastic/package-spec/code/go/internal/errors"
+	"github.com/elastic/package-spec/code/go/internal/spectypes"
 )
 
-func loadItemContent(fsys fs.FS, itemPath, mediaType string) ([]byte, error) {
-	itemData, err := fs.ReadFile(fsys, itemPath)
+func loadItemSchema(fsys fs.FS, path string, contentType *spectypes.ContentType) ([]byte, error) {
+	data, err := fs.ReadFile(fsys, path)
 	if err != nil {
-		return nil, errors.Wrap(err, "reading item file failed")
+		return nil, ve.ValidationErrors{errors.Wrap(err, "reading item file failed")}
+	}
+	if contentType != nil && contentType.MediaType == "application/x-yaml" {
+		return convertYAMLToJSON(data)
 	}
+	return data, nil
+}
 
-	// There might be a situation in which we'd like to keep a directory without any file content under version control.
-	// Usually it can be solved by adding the .empty file. There is no media type defined for this file.
-	// It's expected of the file with media type defined not to be empty - the file can be marked as non-required.
-	if len(itemData) == 0 && mediaType != "" {
-		return nil, errors.New("file is empty, but media type is defined")
+func validateContentType(fsys fs.FS, path string, contentType spectypes.ContentType) error {
+	switch contentType.MediaType {
+	case "application/x-yaml":
+		v, _ := contentType.Params["require-document-dashes"]
+		requireDashes := (v == "true")
+		if requireDashes {
```

Comment:
FYI I remember there was a problem with Kibana validating YML ingest pipelines. It failed when a document didn't have `---`. It shouldn't affect validation if moved here, right?

Reply:
Yes, this is still checked. We have a test (`missing_pipeline_dashes`) that would fail otherwise.